utils/crypto/bls/signer/rpcsigner/client.go

```diff
@@ -0,0 +1,87 @@
+// Copyright (C) 2019-2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package rpcsigner
+
+import (
+	"context"
+	"errors"
+
+	"google.golang.org/grpc"
+
+	"github.com/ava-labs/avalanchego/utils/crypto/bls"
+
+	pb "github.com/ava-labs/avalanchego/proto/pb/signer"
+)
+
+var (
+	_                   bls.Signer = (*Client)(nil)
+	ErrorEmptyPublicKey            = errors.New("empty public key")
+	ErrorEmptySignature            = errors.New("empty signature")
+)
+
+type Client struct {
+	client pb.SignerClient
+	conn   *grpc.ClientConn
+	pk     *bls.PublicKey
+}
+
+func NewClient(ctx context.Context, conn *grpc.ClientConn) (*Client, error) {
+	client := pb.NewSignerClient(conn)
+
+	pubkeyResponse, err := client.PublicKey(ctx, &pb.PublicKeyRequest{})
+	if err != nil {
+		return nil, err
+	}
+
+	pkBytes := pubkeyResponse.GetPublicKey()
+
+	if pkBytes == nil {
+		return nil, ErrorEmptyPublicKey
+	}
+
+	pk, err := bls.PublicKeyFromCompressedBytes(pkBytes)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Client{
+		client: client,
+		conn:   conn,
+		pk:     pk,
+	}, nil
+}
+
+type signatureResponse interface {
+	GetSignature() []byte
+}
+
+func getSignatureFromResponse[T signatureResponse](resp T) (*bls.Signature, error) {
+	signature := resp.GetSignature()
+	if len(signature) == 0 {
+		return nil, ErrorEmptySignature
+	}
+
+	return bls.SignatureFromBytes(signature)
```

Comment:
I'm not sure why we distinguish between invalid and empty signatures.
```suggestion
	signature := resp.GetSignature()
	return bls.SignatureFromBytes(signature)
```
(which tbh makes me wonder if we even need this helper)